setup.py

```diff
@@ -14,16 +31,17 @@
     url="https://handbook.golem.network/yapapi/",
     download_url="https://github.com/golemfactory/yagna-requests",
     packages=setuptools.find_packages(),
-    install_requires=[
-        "requests==2.25.1",
-        "requests-unixsocket==0.2.0",
-        "httpx==0.18.2",
-        "yapapi-service-manager @ git+https://github.com/golemfactory/yapapi-service-manager.git",
-    ],
-    tests_require=[
-        "pytest==6.2.3",
-        "pytest-asyncio==0.15.1",
-    ],
+
+    #   NOTE: all requirements are in "extras", because there are no common dependencies
+    #         for the requestor and provider side, so the library has to be installed either
+    #         as [provider] or as [requestor]
+    install_requires=[],
```

Comment:
So installing this lib without any of `requestor | provider` won't install deps at all :thinking: 
Can we make one of above options default? (whichever seems more appropriate)
...or can we make the installation breaks if :point_up: isn't specified?

Reply:
Yes, no deps will be installed.
This still **might** be somehow useful, because most of `yagna_requests.serializable_request` will still work, but I don't think this is good enough.

I looked for options to create a "default" extra and found this discussion:
https://discuss.python.org/t/adding-a-default-extra-require-environment/4898/124
I read only a small part of it, but the summary seems to be "this makes some sense, but currently is not possible".

About breaking the installation - that's probably possible in some ugly hackish way and not possible in any other way. I'm against the ugly hackish solution.